File: feature_ui/src/main/java/com/wang/mytest/ui/view/recyclerview/RecyclerViewActivity.java
package com.wang.mytest.ui.view.recyclerview;

import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;

import com.wang.mytest.ui.R;
import com.wang.mytest.ui.view.recyclerview.icon.RecyclerViewIconFragment;
import com.wang.mytest.ui.view.recyclerview.simple.RecyclerViewSimpleFragment;

public class RecyclerViewActivity extends AppCompatActivity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_recycler_view);

        Toolbar toolbar = findViewById(R.id.recycler_view_toolbar);
        setSupportActionBar(toolbar);

        if (savedInstanceState == null) {
            getSupportFragmentManager().beginTransaction()
                    .add(R.id.recycler_view_container, new RecyclerViewSimpleFragment())
                    .commitAllowingStateLoss();
        }
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.recycler_view, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        final int itemId = item.getItemId();
        if (itemId == R.id.options_simple) {
//            getSupportFragmentManager().beginTransaction()
//                    .replace(R.id.recycler_view_container, new RecyclerViewSimpleFragment(), "simple")
//                    .commitAllowingStateLoss();

            while (getSupportFragmentManager().popBackStackImmediate());
        } else if (itemId == R.id.options_icon) {
            getSupportFragmentManager().beginTransaction()
                    .replace(R.id.recycler_view_container, new RecyclerViewIconFragment(), "icon")
                    .addToBackStack("icon")
                    .commitAllowingStateLoss();
        }
        return super.onOptionsItemSelected(item);
    }
}

File: feature_ui/src/main/java/com/wang/mytest/ui/view/recycler_view/RecyclerViewActivity.kt
package com.wang.mytest.ui.view.recycler_view

import android.os.Bundle

import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.ItemTouchHelper
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView

import com.wang.mytest.apt.annotation.Route
import com.wang.mytest.R
import com.wang.mytest.common.toast

/**
 * 参考：
 * https://blog.csdn.net/wangcheng_/article/details/80753216
 */
@Route(path = "/activity/ui/recycler_view", title = "RecyclerView")
class RecyclerViewActivity : AppCompatActivity() {

    private lateinit var recyclerView: RecyclerView
    private lateinit var myAdapter: AdapterNormal

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_recycler_view)

        recyclerView = findViewById(R.id.recycler_view)
        recyclerView.layoutManager = LinearLayoutManager(this, LinearLayoutManager.VERTICAL, false)
        recyclerView.addItemDecoration(ItemDecorationFlag())
        recyclerView.adapter = AdapterNormal().apply {
            myAdapter = this
            data = MutableList(50) { i -> "item-$i" }
            itemClickListener =  { _, position ->
                toast("click position = $position")
            }
            itemDeleteListener = { _, position ->
                toast("delete position = $position")
            }
        }

        ItemTouchHelper(SimpleCallbackNormal(myAdapter)).attachToRecyclerView(recyclerView)
    }
}